### PR TITLE
Config option to disable automatic repo watching

### LINKED ISSUE #653: Owner team automatically watching new repo's
When creating a new repo owned by an organization createRepository() calles addRepository() for the Owner team. addRepository() sets permissions for the owners but also sets watches for them. These watches are not always desired.

Please make the Owner team watches upon repo creation optional, either globally or per organization. Perhaps as part of #145 

Also, when adding a new member to an organisation Owner team, AddTeamMember() does not set watches for the new team member. That is inconsistent and probably a bug.

### PR DESCRIPTION
**Pull request by sebastian-sauer**
Add a new config option AUTO_WATCH_NEW_REPOS to enable / disable the automatic watching of
repos for new repositories and if a user is added to a team.

This could fix #653

### CODE REVIEW

**Comment by lafriks** [comment]
It would be great if that could be added also as per-user option

**Review by adelowo — CHANGES_REQUESTED** [review]
You also need to update `app.ini.sample` https://github.com/go-gitea/gitea/blob/master/custom/conf/app.ini.sample

File: models/org_team.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"strings"
 
 	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
```
Comment (adelowo) [inline models/org_team.go:14]: add a  new line before the "github" import

**Review by kolaente — APPROVED** [review]

**Review by adelowo — APPROVED** [review]

**Comment by kolaente** [comment]
Mssql test failed when merging to master, can someone restart the build?

**Comment by trepmag** [comment]
Hello, I tested the variable set to `AUTO_WATCH_NEW_REPOS=false` as follow:
- when adding a user to the Owner team of an Organization related repositories are being watch by this user
- when creating a repository that repository is getting watch by all user set in the related Organization Owner team

Is thoses use cases are intent to be covered by those use cases?

This was observed with `1.10.3`

**Pull request merged**